lang/syn/src/codegen/program/entry.rs

```diff
@@ -47,14 +47,14 @@ pub fn generate(program: &Program) -> proc_macro2::TokenStream {
         ///
         /// The `entry` function here, defines the standard entry to a Solana
         /// program, where execution begins.
-        pub fn entry<'info>(program_id: &Pubkey, accounts: &'info [AccountInfo<'info>], data: &[u8]) -> anchor_lang::solana_program::entrypoint::ProgramResult {
+        pub fn entry<'info>(program_id: &Pubkey, accounts: &[AccountInfo<'info>], data: &[u8]) -> anchor_lang::solana_program::entrypoint::ProgramResult {
```

Comment:
I think this would be much easier to fix by creating a new [`processor!`](https://github.com/solana-labs/solana/blob/4ee5078e5ffdfff36d3f7920217788e2892c1a85/program-test/src/lib.rs#L183) macro with Anchor's current lifetimes because the same lifetime constraints exist in many places.

Reply:
Modifying the macro only is not enough. The issue comes from the signature expected by `invoke_builtin_function` itself. I think the changes to lifetime and references done in anchor need a bit more work to accommodate all the existing use cases.